src/pages/_app.tsx

```diff
@@ -48,6 +52,14 @@ const App = ({
   // Query client
   const [queryClient] = useState(() => new QueryClient());
 
+  // Authentication
+  const router = useRouter();
+  useEffect(() => {
+    supabase.auth.onAuthStateChange((event) => {
+      if (event == "PASSWORD_RECOVERY") router.push("/account/forgot-password");
```

Comment:
What would happen if someone were to go to the route with an URL?

Reply:
At review time, this is what would happen:
- Not logged in, password recovery not declared: see page but can't interact
- Not logged in, password recovery declared: can use the page to set new password
- Logged in: can use the page to set new password (without going through the email process)

Logged in users should not be able to exploit the Password Reset process to change their password without any form of authentication, so I pushed a fix.